### ISSUE TITLE
librfxcodec is not working?

### ISSUE THREAD

**Issue by metalefty** [neutrinolabs/xrdp#358]
@jsorg71 , I can't get librfxcodec working.  When I connect to xrdp with 24/32bpp, screen black out but the desktop seems to be working under black screen.  When I reconnect to the session with 16bpp, successfully reconnects and I can get the screen.

As same result as with devel branch. I also tried on Debian and FreeBSD but the same result.  I suppose that xrdp's issue.Does anyone succeedto use librfxcodec?

Client: Windows 7/10 mstsc.exe, FreeRDP 1.1.0

xrdp: master f422461
xorgxrdp: a0add6c
librfxcodec: 68412e5

Server OS: CentOS 7
X.Org: see below

```
X.Org X Server 1.17.2
Release Date: 2015-06-16
X Protocol Version 11, Revision 0
Build Operating System:  2.6.32-220.17.1.el6.x86_64
Current Operating System: Linux ip-10-100-100-36.ap-northeast-1.compute.internal 3.10.0-327.10.1.el7.x86_64 #1 SMP Tue F
eb 16 17:03:50 UTC 2016 x86_64
Kernel command line: BOOT_IMAGE=/boot/vmlinuz-3.10.0-327.10.1.el7.x86_64 root=UUID=ef6ba050-6cdc-416a-9380-c14304d0d206
ro console=tty0 console=ttyS0,115200n8 crashkernel=auto console=ttyS0,115200 LANG=en_US.UTF-8
Build Date: 20 November 2015  02:44:25PM
Build ID: xorg-x11-server 1.17.2-10.el7
Current version of pixman: 0.32.6
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
```

[xrdp_log.txt](https://github.com/neutrinolabs/xrdp/files/204107/xrdp_log.txt)
[xrdp-sesman_log.txt](https://github.com/neutrinolabs/xrdp/files/204106/xrdp-sesman_log.txt)

**Comment by moobyfr** [comment]
I saw the same issue some times ago, the rdpDeferredUpdateCallback was filling the logs. I don't remember having a solution for that

**Comment by metalefty** [comment]
Is it working for you now?

**Comment by slavach** [comment]
I have the same problem. My client is FreeRDP. I have tried 1.1 as well as 2.0 development.

xrdp: master 87208db
xorgrdp: a0add6c
librfxcodec: 61f6e92

OS: Ubuntu 16.04

```
X.Org X Server 1.18.3
Release Date: 2016-04-04
[     9.773] X Protocol Version 11, Revision 0
[     9.773] Build Operating System: Linux 3.13.0-85-generic x86_64 Ubuntu
[     9.773] Current Operating System: Linux sx2 4.4.0-18-generic #34-Ubuntu SMP Wed Apr 6 14:01:02 UTC 2016 x86_64
[     9.773] Kernel command line: BOOT_IMAGE=/vmlinuz-4.4.0-18-generic root=/dev/mapper/system-slash ro quiet splash
[     9.773] Build Date: 07 April 2016  09:18:50AM
[     9.773] xorg-server 2:1.18.3-1ubuntu2 (For technical support please see http://www.ubuntu.com/support) 
[     9.773] Current version of pixman: 0.33.6
[     9.773]    Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
```

**Comment by moobyfr** [comment]
@metalefty : what was your initial resolution when starting the session? I've tested again, I have a working configuration for small resolution (1024x768, 1280x1024, 1600x1200), but not large (1920.1080, 1920x1200)

**Comment by metalefty** [comment]
I remember my resolution was 1920x1080 or 1200.  I'll test it again in small resolution.

**Comment by metalefty** [comment]
Resolution was not the problem in my case. It didn't work with 1600x900.

Screen is black but it is working.  I feel like I am a blind person.
I can input commands with keyboard and mouse cursor shape changes when it comes edge of screen.

**Comment by moobyfr** [comment]
Can you paste a log with log level 10 ?

**Comment by metalefty** [comment]
Here.  
[log.txt](https://github.com/neutrinolabs/xrdp/files/267577/log.txt)

**Comment by nktl** [comment]
Same problem, black screen when trying to use librfxcodec ('LAN' setting on the client). Xrdp built with RH-Matic. This seems to be only a problem with Xorg drivers, Xvnc mode seems to work OK (although I am not sure if librfxcodec is enabled there)

**Comment by metalefty** [comment]
The log wasn't log level 10. Log again.

xorgxrdp: f745376
xrdp: a25e5e9

[Xorg.12_log.txt](https://github.com/neutrinolabs/xrdp/files/324703/Xorg.12_log.txt)

**Comment by coldfire84** [comment]
I'm also getting this with Debian 8.6; 24bit or 32bit colour with librfxcodec = no windows / menus etc, just the colour of the logon screen. 16-bit I get the full desktop.

Did you ever get to the bottom of this one?

**Comment by metalefty** [comment]
spiedy said:
> RemoteFx is not ready for use with mstsc yet.
> We have to implement the surface commands.

**Comment by metalefty** [comment]
Fixed by #531 . I'll close this after more testing.

**Comment by metalefty** [comment]
I still have some problem with librfxcodec + Win 10 mstsc.exe but the original issue is solved. Closing.
